### ISSUE TITLE
usbradio adds hardcoded 2000mS delay before setting rxCarrierDetect=0 when in VOX mode

### ISSUE THREAD

**Issue by davidgsd** [AllStarLink/ASL-Live-Build#38]
**Describe the bug**
usbradio's `carrierfrom=vox` mode works very well for detecting when audio is present on the Rx Audio line. This makes it easy to use an HT - or any other radio that does not have a COS output - as a node radio, and for usbradio to then simply and reliably detect when any audio signal (even just ~1mV of background room noise) is present, thus enabling support for a wide variety of radios that don't have an easily accessible COS output. This has been working very well for me, but there is one issue: asterisk/channels/xpmr/xpmr.c defines a hardcoded delay,
`#define XPMR_VOX_HANGTIME 2000`
After transmitting into your node and unkeying your radio, `xpmr.c:PmrRx()` then does a countdown and will not set `pChan->rxCarrierDetect` to 0 until the full 2000mS have elapsed. `chan_usbradio.c:usbradio_read()` then sets `o->rxcarrierdetect` to 0 (cd) if `rxcdtype==CD_XPMR_VOX`.

This results in a full 2 second delay after every transmit before it is considered to have ended and thus causes a very noticeable and awkward silence for all users on connected remote nodes and systems, before a repeater courtesy tone plays and other nodes/users can key up.  

A small delay of maybe 100mS might be understandable, but I can see no reason why the delay should be any longer than that, and it should in any case be configurable with a setting in the usbradio.conf or other .conf file.

I would definitely consider this a bug, since `carrierfrom=vox` mode is not really usable as-is. If this were to be made a configuration parameter, such as adding a new parameter named `voxhangtime` after the `carrierfrom` parameter in usbradio.conf, it could have the same default value as `XPMR_VOX_HANGTIME` (2000 mS), thereby ensuring full backwards-compatibility in case any existing nodes/users actually want that long of a delay, but would enable the value to be reduced to something more reasonable and thereby make the vox mode much more useful.

**To Reproduce**
1. In usbradio,conf set carrierfrom=vox, ctcssfrom=no, rxdemod=speaker.
2. Disconnect COS line (if present) from your node radio or USB interface. Set the squelch control on your radio so there is no noise on the Rx Audio line when you are not transmitting into your node.
3. Transmit audio into your node. Your transmission will then go out to any connected nodes, and if you watch the status on a remote node you'll see that your node shows as keyed for a full 2 seconds after you actually unkeyed.

**Expected behavior**
`XPMR_VOX_HANGTIME` should be a configuration parameter instead of a `#define`, so it can be set in a .conf file to any desired value, or at least within a range of ~ 20 - 2000 mS.

**Screenshots**
I could provide a video if desired of me talking into my node, unkeying, and a remote node monitor page then showing I'm still keyed for 2 additional seconds.

**Additional context**
I am very impressed with how well ASL and usbradio works at reliably detecting the presence of even a quiet carrier on the Rx audio line. It does this very reliably for me with no false triggers and no false dropouts. The long 2 second delay however pretty much makes the whole feature unusable. Thus it seems that making this configurable should be a simple change that will add a lot of value to ASL in a significant range of use cases. I believe many people would like to build their own nodes using an HT for the node radio and thereby get away from the commercial nodes which often are high-priced and/or have low-quality noisy RF modules and proprietary software, but the COS requirement is a challenge because it requires either modifying an HT, using a more expensive mobile radio having a COS output, or using the usbradio vox function and dealing with the 2 second delay after every Tx (which also affects all other connected nodes).

I've been a EE and software engineer doing C in telecom and satcomm systems from 1993-2010 (then moved to PHP & SQL) and I would be happy to do a pull request and test this change if that would help. But maybe someone else is more familiar with the code and could take care of this more quickly. Seems like it should be a pretty simple change. I have plenty of time available and would be happy to help and could probably get a pull request completed in the next week.

Thanks, 
David NR9V

**Comment by davidgsd** [comment]
Update: I was able to follow the instructions on https://github.com/AllStarLink/ASL-Asterisk#compiling and get the code to compile on my RPi4. I changed the XPMR_VOX_HANGTIME to 500 mS and verified that everything now works very well. My transmits now end much sooner after I unkey, and I see no false drops in carrier detect even when not talking. Since the code is easy to compile this was an easy fix for me, but hopefully for the benefit of other users of usbradio vox this could be made configurable in a future ASL release. I would be happy to make a pull request and set up a new cfg if that would help.
